File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    const answer = [];
    const stack = [];
    
    for (const s of score) {
        if (stack.length < k) {
            stack.push(s);
            stack.sort((a, b) => b - a);
            answer.push(Math.min(...stack));
            continue;
        }
        
        if (Math.min(...stack) < s) {
            stack.pop();
            stack.push(s);
            stack.sort((a, b) => b - a);
        }
        
        answer.push(Math.min(...stack));
    }
    
    return answer;
}

File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.py
# 매일 발표된 명예의 전당의 최하위 점수를 return
import heapq

def solution(k, score):
    answer = []
    
    heap = []
    for s in score:
        if len(heap) < k:
            heapq.heappush(heap, s)
            answer.append(min(heap))
            continue
        
        if min(heap) < s:
            heapq.heappop(heap)
            heapq.heappush(heap, s)
            
        answer.append(min(heap))
    return answer